### PR TITLE
 [MSG] Update Conversation schema and add message pagination

### PR DESCRIPTION
**Pull request by sarahscott**
Part of https://github.com/artsy/collector-experience/issues/294

- [x] Add message pagination
- [x] Figure out why `FlatList` component looks messed up for conversations with a lot of messages (maybe switch back to `ListView`?)
- [x] Make sure synced schema corresponds with actual Metaphysics `master` schema (currently it's from my branch)
- [x]  Tests

Backward pagination (if we go down that route) will happen in a different pr.

simulator screen shot jun 16 2017 4 52 37 pm

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  No CHANGELOG added.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please assign someone to merge this PR, and optionally include people who should review.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/components/inbox/conversations/index.tsx [hunk 1]
```diff
@@ -45,7 +45,13 @@ export class Conversations extends React.Component<Props, State> {
   }
 
   get conversations() {
-    return this.props.me.conversations.edges.map(a => a.node) || []
+    // It's currently possible for a conversation to be message-less in impulse, in which case we shouldn't show it
```
Comment (sarahscott) [inline src/lib/components/inbox/conversations/index.tsx]: Wondering if metaphysics should handle this - i.e. just not send a conversation at all if it doesn't contain messages. But maybe we should safeguard the client anyway?

File: src/lib/containers/conversation.tsx [hunk 2]
```diff
@@ -51,42 +54,58 @@ const MessagesList = styled(FlatList)`
...
+    })
 
     return (
       <Container>
         <Header>
           <HeaderTextContainer>
             <BackButtonPlaceholder source={chevron} />
             <SmallHeadline>{partnerName}</SmallHeadline>
-            <MetadataText>Info</MetadataText>
+            <PlaceholderView />
```
Comment (sarahscott) [inline src/lib/containers/conversation.tsx]: We aren't doing an Info button yet but this placeholder is very handy.

File: src/lib/Components/Inbox/Conversations/Message.tsx [hunk 3]
```diff
@@ -59,7 +66,8 @@ export default class Message extends React.Component<Props, any> {
             <SenderName>{this.props.message.senderName}</SenderName>
             <MetadataText>{this.props.message.time}</MetadataText>
           </Header>
-          <BodyText>{this.props.message.body}</BodyText>
+          {this.props.artworkPreview && <ArtworkPreviewContainer>{this.props.artworkPreview}</ArtworkPreviewContainer>}
+          <BodyText>{this.props.message.body.split("\n\nAbout")[0]}</BodyText>
```
Comment (sarahscott) [inline src/lib/Components/Inbox/Conversations/Message.tsx]: This is just a really rudimentary clean-up of the initial message so it's easier to see what things should actually look like. We'll eventually do real message sanitizing in metaphysics.

File: src/lib/Components/Inbox/Conversations/index.tsx [hunk 4]
```diff
@@ -45,7 +45,13 @@ export class Conversations extends React.Component<Props, State> {
   }
 
   get conversations() {
-    return this.props.me.conversations.edges.map(a => a.node) || []
+    // It's currently possible for a conversation to be message-less in impulse, in which case we shouldn't show it
```
Comment (mennenia) [inline src/lib/Components/Inbox/Conversations/index.tsx]: :+1:

**Pull request merged**